windows/src/desktop/setup/RunTools.pas

```diff
@@ -642,7 +642,7 @@ procedure TRunTools.ConfigFirstRun(StartKeyman,StartWithWindows,CheckForUpdates,
       s := '-s -install-tips-for-packages ';
       for pack in FInstallInfo.Packages do
       begin
-        if pack.ShouldInstall then
+        if pack.ShouldInstall and Assigned(pack.InstallLocation) then
```

Comment:
In what sort of conditions would the former be true bot not the latter?  My intuitive understanding of this line is:

"If the package should be installed and the install location has been assigned, then..."

But from the PR's description:

> 1. RunTools.pas: don't attempt to install TIPs if package is not also being installed.

Feels like a bit of a mismatch.

Reply:
There are scenarios where `pack.ShouldInstall` may never be set earlier in the code, because no install source is found (aka Location).

Reply:
So... `pack.ShouldInstall` defaults to `true` unless set?  Otherwise, I don't see how that answers the question.

Reply:
Yep -- it can be turned off by the user in Install Options dialog, so it's really a user flag.